src/model_config_tests/qa/test_access_esm1p5_config.py

```diff
@@ -191,3 +197,44 @@ def test_mppnccombine_fast_collate_exe(self, config):
             assert isinstance(config["collate"]["mpi"], bool), error_field_incorrect(
                 "collate.mpi", "config.yaml", "true or false"
             )
+
+    def test_cice_configuration_icefields_nml_in_ice_history_nml(
+        self, config, control_path
+    ):
+        # Find CICE sub-model control path
+        model_name = None
+        for sub_model in config["submodels"]:
+            if sub_model["model"] == "cice":
+                model_name = sub_model["name"]
+        assert model_name
+        cice_control_path = control_path / model_name
+
+        icefields_nml_error_msg = (
+            "Expected CICE configuration to have ice_history.nml that "
+            "contains an icefields_nml namelist. This is to keep icefields "
+            "namelist separate from the cice_in.nml to allow simpler changes."
+        )
+
+        # Check ice_history.nml exists
+        ice_history_nml_path = cice_control_path / ICE_HISTORY_NML_FNAME
+        assert ice_history_nml_path.is_file(), icefields_nml_error_msg
+
+        # Check icefields_nml in ice_history.nml
+        ice_history_nml = f90nml.read(ice_history_nml_path)
+        assert ICEFIELDS_NML_NAME in ice_history_nml, icefields_nml_error_msg
+
+        # Check icefields_nml not in cice_in.nml
+        cice_in_path = cice_control_path / CICE_IN_NML_FNAME
+        assert cice_in_path.is_file()
+        cice_in = f90nml.read(cice_in_path)
+        assert ICEFIELDS_NML_NAME not in cice_in, icefields_nml_error_msg
```

Comment:
```suggestion
        assert ICEFIELDS_NML_NAME not in cice_in, icefields_nml_wrong_file_error_msg
```
Sorry to be a pedant, this should have a different error message. e.g.:

"icefields_nml namelist found in cice_in.nml. This should only be in ice_history.nml to prevent duplication."

Reply:
Thanks, that does make it more clear! I've added that in 0e91aec.